app/controllers/assignment_invitations_controller.rb

```diff
@@ -4,8 +4,7 @@ class AssignmentInvitationsController < ApplicationController
   before_action :check_user_not_previous_acceptee, only: [:show]
 
   def accept_invitation
-    users_assignment_repo = invitation.redeem_for(current_user)
-
+    users_assignment_repo = invitation.redeem_for(current_user, params[:repo_name_suffix])
     if users_assignment_repo.present?
       redirect_to successful_invitation_assignment_invitation_path
     else
```

Comment:
I reused this endpoint.

When there is a GitHub repo already exists with the same name:
- If `params[:repo_name_suffix]` is `nil`, we add the user to the repo as a collaborator.
- Else if `params[:repo_name_suffix]` is not `nil`, we create a new repo with `repo_name_suffix` and add the user to the repo.